session/session_test.go

```diff
@@ -5828,7 +5828,8 @@ func (s *testSessionSuite) TestTiDBReadStaleness(c *C) {
 	c.Assert(err, NotNil)
 	err = tk.ExecToErr("set @@tidb_read_staleness='foo'")
 	c.Assert(err, NotNil)
-	tk.MustExec("set @@tidb_read_staleness=''")
+	err = tk.ExecToErr("set @@tidb_read_staleness='100'")
```

Comment:
This case will fail in CI, I think it should be checked in `sysvar.go`.

Reply:
If this only supports negative values, we should test one of those too.

Reply:
@morgo But why `set @@tidb_read_staleness='100'` can succeed now? Because the MaxValue is 0, shouldn't the validation will automatically be performed?

Reply:
Yes it should. We don't have any other examples where the range is been `-val` and `0` though, maybe there is a bug?

My point was that we should test that `-100` is tested, as an example of an expected valid value.

Reply:
There is an example `set @@tidb_read_staleness='-5'`. If there is a bug, the range should be checked by ourselves.

Reply:
Ahh yes! The MySQL behavior is to autoconvert to the closest permitted value + raise a warning. This is the case since https://github.com/pingcap/tidb/pull/25686 It is only wrong typed values that will error.

If you modify the test to read back the value, it should be zero?

Reply:
I think yes.

Reply:
Any modification is needed for this PR? @morgo